src/components/token/TokenBalances.js

```diff
@@ -0,0 +1,327 @@
+     *         Sort is an array with two strings, The value is given by ElasticSearch and it is passed back when we want to change page
+     * hasAfter: Indicates if a next page exists
+     * hasBefore: Indicates if a previous page exists
+     * sortBy: Which field to sort (uid, name, symbol)
+     * order: If sorted field must be ordered asc or desc
+     * page: Current page. Used to know if there is a previous page
+     * pageSearchAfter: Calculates the searchAfter param that needs to be passed to explorer-service in order to get the next/previous page.
+     *                  We use this array to store already-calculated values,
+     *                  so we do not need to recalculate them if user is requesting an already-navigated page.
+     * loading: Initial loading, when user clicks on the Tokens navigation item
+     * isSearchLoading: Indicates if search results are being retrieved from explorer-service
+     * calculatingPage: Indicates if next page is being retrieved from explorer-service
+     * error: Indicates if an unexpected error happened when calling the explorer-service
+     * tokenBalanceInformationError: Indicates if an unexpected error happened when calling the token balance information service 
+     * maintenanceMode: Indicates if explorer-service or its downstream services are experiencing problems. If so, maintenance mode is enabled as
+     *                  a "circuit breaker" to remove additional load until the team fixes the problem
+     */
+    this.state = {
+      tokenId: '00',
+      tokenBalances: [],
+      hasAfter: false,
+      hasBefore: false,
+      sortBy: 'total',
+      order: 'desc',
+      page: 1,
+      pageSearchAfter: [],
+      loading: false,
+      isSearchLoading: false,
+      calculatingPage: false,
+      error: false,
+      tokenBalanceInformationError: false,
+      maintenanceMode: this.props.maintenanceMode,
+      transactionsCount: 0,
+      addressesCount: 0,
+    };
+  }
+
+  componentDidMount = async () => {
+    if (this.state.maintenanceMode) {
+      return;
+    }
+
+    // 'Click' on search to make the first query
+    const queryParams = this.pagination.obtainQueryParams();
+
+    await this.setState({
+      sortBy: get(queryParams, 'sortBy', this.state.sortBy),
+      order: get(queryParams, 'order', this.state.order),
+      loading: true,
+    });
+
+    await this.performSearch();
+
+    this.setState({
+      loading: false,
+    });
+  }
+
+  /**
+   * Call explorer-service to get list of token balances for a given tokenId
+   *
+   * @param {*} searchAfter Parameter needed by ElasticSearch for pagination purposes
+   * @returns tokens
+   */
+  getTokenBalances = async (searchAfter) => {
+    const tokenBalancesRequest = await tokensApi.getBalances(this.state.tokenId, this.state.sortBy, this.state.order, searchAfter);
+    this.setState({
+      error: get(tokenBalancesRequest, 'error', false),
+    });
```

Comment:
I think setting this every time will trigger unnecessary re-renders in the component.

> The default behavior is to re-render on every state change, and in the vast majority of cases you should rely on the default behavior.

https://reactjs.org/docs/react-component.html#shouldcomponentupdate

Unless we turn this component into a `PureComponent`, which would perform a shallow comparison in the state before triggering a re-render.

But maybe we should just set the state in case there is actually an error.